sftpcloudfs/main.py

```diff
@@ -359,17 +359,29 @@ def __init__(self):
             keystone_keys = ('auth_version', 'region_name', 'tenant_separator', 'domain_separator', 'service_type', 'endpoint_type')
             options.keystone = dict((key, getattr(options, key)) for key in keystone_keys)
 
-        if options.uid and isinstance(options.uid, basestring):
+        if options.uid:
             try:
-                options.uid = pwd.getpwnam(options.uid).pw_uid
-            except KeyError:
-                parser.error("uid: Failed to get %s's uid" % options.uid)
-
-        if options.gid and isinstance(options.gid, basestring):
+                options.uid = int(options.uid)
+            except ValueError:
+                if isinstance(options.uid, basestring):
```

Comment:
uid/gid will always be a string now? Is this necessary?

We try parsin an int, if it fils we try to find the id assuming user/group name, if it fails: report error.

If this doesn't make sense I'm happy to merge the PR as it is.

Reply:
I assume this is because `OptionParser.add_option()` default type is `string`, so I guess I will always be a string now.

But I can change the logic it to : 
1. assuming this will be a uid/gid name
2. if not, convert numerical uid/gid string as int and try again
3. if it fails, report error

What do you think ?

Reply:
`adduser` tool uses this to verify an username: `NAME_REGEX="^[a-z][-a-z0-9]*\$"`

I think you have to check if is numeric first and use it verbatim because a username can't be a numeric value. If is not numeric, then try tto convert the user/group to uid/gid.

Reply:
Or something like this ?

```
if options.uid:
    try:
        options.uid = pwd.getpwnam(options.uid).pw_uid
    except KeyError:
        try:
            options.uid = pwd.getpwuid(int(options.uid))
        except KeyError:
            parser.error("uid: Invalid uid: %s" % options.uid)
```

- `options.uid` i always a string
- Check passwd db with user name (str mandatory here)
- If fail, check passwd db with user id (int mandatory here)
- If both fail, report error

Reply:
The daemon context expect an integer.

The user can provide either a numeric value (old behaviour) or a user name. If the option is a numeric, it can't be a user. So we should test for numeric first, if it fails user, finally report error.

Something like this (not tested!):
```python
if options.uid:
    try:
        options.uid = int(options.uid)
    except ValueError:
        try:
            options.uid = pwd.getpwnam(options.uid).pw_uid
        except KeyError:
            parser.error("uid: Invalid uid: %s" % options.uid)
```

Reply:
To be fair, the issue here is more about using a field for two things. If the conf token is `uid` I would expect to provide a numeric. If it was `user`, I would expect to provide a string.

So we could revert to let uid/gid be numerics and add new parameters to provide user/group; if both uid and user are provided, report error an exit; otherwise use the type of input provided.

I leave it to you. Both approaches are OK; but in case we accept both numeric id or a user, I think we need to parse it as I explained in my previous comment.

Reply:
> To be fair, the issue here is more about using a field for two things. If the conf token is `uid` I would expect to provide a numeric. If it was `user`, I would expect to provide a string.

Agree on that.
Let's stick to a common field for now, but I'll don't mind to change it if we get people confused :)